src/classes/RD2_NamingService.cls

```diff
@@ -203,51 +211,220 @@ public with sharing class RD2_NamingService {
+    }
 
-        String nameSuffix = ' ' + formatAmount(rd) + ' - ' + System.Label.RecurringDonationNameSuffix;
+    /**
+     * @return The currently configured RD naming format string
+     */
+    private String getCurrentNamingFormat() {
 
-        String donorName = donor == null
-            ? ''
-            : donor.Name.abbreviate(MAX_NAME_LENGTH - nameSuffix.length());
+        String format = UTIL_CustomSettingsFacade.getRecurringDonationsSettings().RecurringDonationNameFormat__c;
+        if (format == null) {
+            format = RD_NAMING_OPTION_1; // the default format
+        }
 
-        return donorName + nameSuffix;
+        return format;
     }
 
-    /***
-    * @description Formats amount used to construct the name
-    * @param rd Recurring Donation record
-    * @return String Formatted amount
-    */
-    private String formatAmount(npe03__Recurring_Donation__c rd) {
-        String formattedAmount = rd.npe03__Amount__c == null ? '' : rd.npe03__Amount__c.format();
+    /**
+     * @description Build out the Recurring Donation name based on the define naming format
+     */
+    private class RecurringDonationNameBuilder {
+
+        private npe03__Recurring_Donation__c rd;
+        private String nameFormat;
+        private Map<Id, RD2_Donor.Record> donorById;
+
+        private String rdName;
+
+        /**
+         * @description Constructor
+         * @param rd Recurring Donation
+         */
+        public RecurringDonationNameBuilder(npe03__Recurring_Donation__c rd) {
+            this.rd = rd;
+        }
+
+        /**
+         * @param nameFormat The RD Naming format to use
+         * @return current instance
+         */
+        public RecurringDonationNameBuilder withNameFormat(String nameFormat) {
+            this.nameFormat = nameFormat;
+            return this;
+        }
 
-        if (formattedAmount.endsWith('.00')) {
-            formattedAmount = formattedAmount.removeEnd('.00');
-        } else if (formattedAmount.endsWith('.0')) {
-            formattedAmount = formattedAmount.removeEnd('.0');
+        /**
+         * @param donorById Map of Donor Account or Contact information by RD Id
+         * @return current instance
+         */
+        public RecurringDonationNameBuilder withDonorsById(Map<Id, RD2_Donor.Record> donorById) {
+            this.donorById = donorById;
+            return this;
         }
 
-        String currencySymbol = userDefaultCurrencySymbol;
-        if (isMultiCurrencyEnabled) {
-            String code = (String) rd.get(CURRENCY_ISO_CODE_FIELD);
-            currencySymbol = UTIL_Currency.getInstance().getCurrencySymbol(code);
+        /**
+         * @description Generate the RD name
+         * @return Recurring Donations Name
+         */
+        public String build() {
+            rdName = this.nameFormat;
+
+            rdName = replaceTokenValues();
+            rdName = replaceFieldValues();
+            rdName = replaceDonorName();
```

Comment:
`rdName =` part is unnecessary since the variable is global and is being modified in the help functions